.github/workflows/gradle.yml

```diff
@@ -15,17 +12,25 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up JDK 1.8
-      uses: actions/setup-java@v1
-      with:
-        java-version: 1.8
-    - name: Grant execute permission for gradlew
-      run: chmod +x gradlew
-    - name: Build with Gradle
-      run: ./gradlew build
-    - name: Unit tests results
-      uses: actions/upload-artifact@v2
-      with:
-        name: unit-tests-results
-        path: dynatrace/build/reports/tests/test/ 
+      - uses: actions/checkout@v2
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+      - name: Compile
+        run: ./gradlew assemble
+      - name: Test
+        run: ./gradlew build
```

Comment:
is there a reason why we don't use `./gradlew check` here instead of build? AFAIK `build` will run `assemble` and `check` one after the other, but will probably ignore the `assemble` step in this case as you specified it explicitly before anyway.

Reply:
In this case it's equivalent since `assemble` is executed just before but since the step is named "test", it makes sense to only run `check` - will adapt.